src/pages/index.js

```diff
@@ -3,32 +3,99 @@ import { FormValidator } from "../componets/FormValidator.js";
 import Section from "../componets/Section.js";
 import PopupWithImage from "../componets/PopupWithImage.js";
 import PopupWithForm from "../componets/PopupWithForm.js";
+import PopupWithConfirmation from "../componets/PopupWithConfirmtaion.js";
 import UserInfo from "../componets/UserInfo.js";
-import { initialCards, formSettings } from "../utils/constants.js";
+import { formSettings, token } from "../utils/constants.js";
 import "./index.css";
+import Api from "../componets/Api.js";
+
+// Profile consts
+const profile = document.querySelector(".profile");
+const profileName = profile.querySelector(".profile__name");
+const profileDecsription = profile.querySelector(".profile__description");
+const profileAvatar = profile.querySelector(".profile__avatar");
+
+const profileInfo = new UserInfo({
+  name: profileName,
+  description: profileDecsription,
+  avatar: profileAvatar,
+});
+
+// Cards and Cardlist
+const cardListLocation = document.querySelector(".cards");
+
+function deleteCard(cardId, card) {
+  return api
+    .deleteCard(cardId)
+    .then((res) => {
+      if (res.ok) {
+        card.remove();
+        return Promise.resolve();
+      } else {
+        return Promise.reject(res.status);
+      }
+    })
+    .catch((err) => {
+      console.error(`ERROR: Failed to delete card: ${err}`);
+    });
+}
+
+function handleCardLike(isLiked, cardId) {
+  if (isLiked) {
+    return api.removeCardLike(cardId).then((res) => {
+      if (res.ok) {
+        return Promise.resolve();
+      } else {
+        return Promise.reject(`Failed to remove like: ${res.status}`);
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call. 

**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.